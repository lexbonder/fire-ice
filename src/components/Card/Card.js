import React from 'react';
import PropTypes from 'prop-types';


export const Card = ({houseInfo}) => {
  const {
    name,
    founded,
    seats,
    titles,
    coatOfArms,
    ancestralWeapons,
    words
  } = houseInfo;

const seatsToRender = () => {
  return seats.map( (seat, index) => <h3 key={index}>{seat}</h3>)
}

const titlesToRender = () => {
  return titles.map( (title, index) => <h3 key={index}>{title}</h3>)
}

const weaponsToRender = () => {
  return ancestralWeapons.map( (weapon, index) => <h3 key={index}>{weapon}</h3>)
}

  return (
    <div className='Card'>
      <h1>{name}</h1>
      <h2>{founded}</h2>
      <h2>Seats</h2>
      {seatsToRender()}
      <h2>Titles</h2>
      {titlesToRender()}
      <h2>{coatOfArms}</h2>
      <h2>Ancestral Weapons</h2>
      {weaponsToRender()} 
      <h2>{words}</h2>
    </div>
  );
};

const { arrayOf, shape, string } = PropTypes;

Card.propTypes = {
  houseInfo: shape({
    url: string,
    name: string,
    coatOfArms: string,
    words: string,
    titles: arrayOf(string),
    seats: arrayOf(string),
    founded: string,
    ancestralWeapons: arrayOf(string)
  })
};