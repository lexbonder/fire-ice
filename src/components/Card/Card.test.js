import React from 'react';
import { shallow } from 'enzyme';
import { Card } from './Card';




describe('Card', () => {
  let mockHouse;

  beforeEach(() => {
    mockHouse = {
      url: 'https://www.anapioficeandfire.com/api/houses/81',
      name: 'House Corbray of Heart\'s Home',
      coatOfArms: 'Three black ravens in flight, holding three red hearts, on a white field(Argent, three ravens volant sable, each clutching in their claws a heart gules)',
      words: '',
      titles: [
        'Lord of Heart\'s Home',
        'King of the Fingers (historical)'
      ],
      seats: [
        'Heart\'s Home'
      ],
      founded: '',
      ancestralWeapons: [
        'Lady Forlorn'
      ]
    }
  })

  it('should match the snapshot', () => {    
    const wrapper = shallow(<Card houseInfo={mockHouse} />)
    expect(wrapper).toMatchSnapshot()
  })
})