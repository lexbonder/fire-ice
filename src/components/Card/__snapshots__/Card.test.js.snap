// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    houseInfo={
        Object {
            "ancestralWeapons": Array [
              "Lady Forlorn",
            ],
            "coatOfArms": "Three black ravens in flight, holding three red hearts, on a white field(Argent, three ravens volant sable, each clutching in their claws a heart gules)",
            "founded": "",
            "name": "House Corbray of Heart's Home",
            "seats": Array [
              "Heart's Home",
            ],
            "titles": Array [
              "Lord of Heart's Home",
              "King of the Fingers (historical)",
            ],
            "url": "https://www.anapioficeandfire.com/api/houses/81",
            "words": "",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1>
          House Corbray of Heart's Home
</h1>,
        <h3 />,
        <h3>
          Heart's Home
</h3>,
        <h3>
          Lord of Heart's Home
          King of the Fingers (historical)
</h3>,
        <h3>
          Three black ravens in flight, holding three red hearts, on a white field(Argent, three ravens volant sable, each clutching in their claws a heart gules)
</h3>,
        <h3>
          Lady Forlorn
</h3>,
        <h3 />,
      ],
      "className": "Card",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "House Corbray of Heart's Home",
        },
        "ref": null,
        "rendered": "House Corbray of Heart's Home",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "",
        },
        "ref": null,
        "rendered": "",
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Heart's Home",
          ],
        },
        "ref": null,
        "rendered": Array [
          "Heart's Home",
        ],
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Lord of Heart's Home",
            "King of the Fingers (historical)",
          ],
        },
        "ref": null,
        "rendered": Array [
          "Lord of Heart's Home",
          "King of the Fingers (historical)",
        ],
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Three black ravens in flight, holding three red hearts, on a white field(Argent, three ravens volant sable, each clutching in their claws a heart gules)",
        },
        "ref": null,
        "rendered": "Three black ravens in flight, holding three red hearts, on a white field(Argent, three ravens volant sable, each clutching in their claws a heart gules)",
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Lady Forlorn",
          ],
        },
        "ref": null,
        "rendered": Array [
          "Lady Forlorn",
        ],
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "",
        },
        "ref": null,
        "rendered": "",
        "type": "h3",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            House Corbray of Heart's Home
</h1>,
          <h3 />,
          <h3>
            Heart's Home
</h3>,
          <h3>
            Lord of Heart's Home
            King of the Fingers (historical)
</h3>,
          <h3>
            Three black ravens in flight, holding three red hearts, on a white field(Argent, three ravens volant sable, each clutching in their claws a heart gules)
</h3>,
          <h3>
            Lady Forlorn
</h3>,
          <h3 />,
        ],
        "className": "Card",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "House Corbray of Heart's Home",
          },
          "ref": null,
          "rendered": "House Corbray of Heart's Home",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "",
          },
          "ref": null,
          "rendered": "",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Heart's Home",
            ],
          },
          "ref": null,
          "rendered": Array [
            "Heart's Home",
          ],
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Lord of Heart's Home",
              "King of the Fingers (historical)",
            ],
          },
          "ref": null,
          "rendered": Array [
            "Lord of Heart's Home",
            "King of the Fingers (historical)",
          ],
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Three black ravens in flight, holding three red hearts, on a white field(Argent, three ravens volant sable, each clutching in their claws a heart gules)",
          },
          "ref": null,
          "rendered": "Three black ravens in flight, holding three red hearts, on a white field(Argent, three ravens volant sable, each clutching in their claws a heart gules)",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Lady Forlorn",
            ],
          },
          "ref": null,
          "rendered": Array [
            "Lady Forlorn",
          ],
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "",
          },
          "ref": null,
          "rendered": "",
          "type": "h3",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
